File: app/src/main/java/me/imzack/app/end/injector/module/AppModule.java
package me.imzack.app.end.injector.module;

import android.content.Context;

import me.imzack.app.end.model.DataManager;

import org.greenrobot.eventbus.EventBus;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private final Context mContext;

    //仅能通过App提供的才需要通过构造方法传入
    public AppModule(Context context) {
        mContext = context;
    }

    //AppComponent构造过程中并不会调用以下的provide方法

    @Provides
    @Singleton
    Context provideContext() {
        return mContext;
    }

    @Provides
    @Singleton
    DataManager provideDataManager() {
        return DataManager.getInstance();
    }

    @Provides
    @Singleton
    EventBus provideEventBus() {
        return EventBus.getDefault();
    }
}

File: app/src/main/kotlin/me/imzack/app/end/injector/module/AppModule.kt
package me.imzack.app.end.injector.module

import android.content.Context
import dagger.Module
import dagger.Provides
import org.greenrobot.eventbus.EventBus
import javax.inject.Singleton

//TODO https://www.kotlincn.net/docs/tutorials/android-frameworks.html

//仅能通过App提供的才需要通过构造方法传入
@Module
class AppModule(private val mContext: Context) {

    //AppComponent构造过程中并不会调用以下的provide方法
    @Provides
    @Singleton
    fun provideContext() = mContext

    @Provides
    @Singleton
    fun provideEventBus() = EventBus.getDefault()!!
}
